airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -397,11 +411,18 @@ def _is_datetime_incremental_without_partition_routing(
         return (
             incremental_sync_component_definition is not None
             and bool(incremental_sync_component_definition)
-            and incremental_sync_component_definition.get("type", "")
-            == DatetimeBasedCursorModel.__name__
+            and (
+                incremental_sync_component_definition.get("type", "")
+                in (DatetimeBasedCursorModel.__name__, IncrementingCountCursorModel.__name__)
+            )
             and hasattr(declarative_stream.retriever, "stream_slicer")
             and (
                 isinstance(declarative_stream.retriever.stream_slicer, DatetimeBasedCursor)
+                # IncrementingCountCursorModel is hardcoded to be of type DatetimeBasedCursor
```

Comment:
Can we update the method name to reflect that? I'm not sure what would be the best name but my understanding is that it's basically `_is_concurrent_cursor_incremental_without_partition_routing`